src/Simplex/Messaging/Agent/Client.hs

```diff
@@ -269,6 +274,15 @@ data AgentClient = AgentClient
     agentEnv :: Env
   }
 
+data Worker = Worker {workerId :: Int, doWork :: TMVar (), action :: Async ()}
+
+newWorker :: AgentMonad' m => AgentClient -> (Int -> TMVar () -> m ()) -> m Worker
+newWorker c runWorker = do
+  workerId <- atomically $ stateTVar (workerSeq c) $ \next -> (next, next + 1)
+  doWork <- newTMVarIO ()
+  action <- async $ runWorker workerId doWork
```

Comment:
misses
```
`agentFinally` atomically (TM.delete srv_ ws)
```
?

Reply:
it happens where it is created

Reply:
also, this may delete another worker created for the same srv_